fiftyone.pipeline.core/evidence.js

```diff
@@ -148,11 +148,23 @@ class Evidence {
 
     evidence.add('server.host-ip', request.connection.localAddress.toString());
 
-    // Get querystring data
+    // Parsing URL using new URL constructor
+
+    const protocol = request.protocol || 'http';
+    const hostname = request.hostname || 'localhost'; // Use a default hostname if not available
+    const port = request.socket.localPort || ''; // Port may not be available in some cases
+    const searchParams = request.query || ''; // Assuming query parameters are in req.query
```

Comment:
Is request an [IncomingMessage](https://www.w3schools.com/nodejs/obj_http_incomingmessage.asp) type?  If so it will have the `.url` property that can then be parsed... and then `searchParams` can be taken from `url.URL(request.url).searchParams` according to the docs..  so the previous implementation seems to have been correct...